HW Lesson 1 Alexandr Laptev

Sanuamba3 opened the pull request:
Lesson 1

Sanuamba3 committed:
HW Lesson 1 Alexandr Laptev

Lesson 1

DmitryChitalov commented on Урок 1. Практическое задание/task_1.py:
```diff
@@ -22,10 +22,10 @@ def check_1(lst_obj):
-    lst_to_set = set(lst_obj)  # !!!
-    return lst_to_set  # !!!
+    lst_to_set = set(lst_obj)  #  O(n)
+    return lst_to_set  # O(1)
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_1.py:
```diff
@@ -37,12 +37,12 @@ def check_2(lst_obj):
+        if lst_obj[j] in lst_obj[j+1:]:    # O(n)
+            return False                   # O(1)
+    return True                            # O(1)
 
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_1.py:
```diff
@@ -53,14 +53,14 @@ def check_3(lst_obj):
+    for i in range(len(lst_obj) - 1):        # O(n)
+        if lst_copy[i] == lst_copy[i+1]:     # O(1)
+            return False                     # O(1)
+    return True                              # O(1)
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_2.py:
```diff
@@ -14,4 +14,30 @@
+        if i < minimal:  # O(1)
+            minimal = i  # O(1)
+    return minimal       # O(1)
+
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_2.py:
```diff
@@ -14,4 +14,30 @@
+
+test1 = [10300,5,367]
+test2 = [34,234,3,34,23,4,5333]
+
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_3.py:
```diff
@@ -17,3 +17,34 @@
+}
+
+print(find_top_3_1(dict))
+print(fint_top_3_2 (dict))
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_4.py:
```diff
@@ -22,3 +22,41 @@
+
+def check_login_1(user): 
+    if user['username'] in users:                                       # O(n)
+        if users[user['username']]['active']:                           # O(n)
```
if users[user['username']]['active']:                           # O(n)
константная

DmitryChitalov commented on Урок 1. Практическое задание/task_4.py:
```diff
@@ -22,3 +22,41 @@
+def check_login_1(user): 
+    if user['username'] in users:                                       # O(n)
+        if users[user['username']]['active']:                           # O(n)
+            if user['password'] == users[user['username']]['password']: # O(n)
```
константная

DmitryChitalov commented on Урок 1. Практическое задание/task_4.py:
```diff
@@ -22,3 +22,41 @@
+    for key, value in users.items():                                    # O(n)
+        if user['username'] == key:                                     # O(n)
+            if value['active']:                                         # O(n)
+                if user['password'] != value['password']:               # O(n)
```
if user['username'] == key:                                     # O(n)
            if value['active']:                                         # O(n)
                if user['password'] != value['password']:               # O(n)

конст.

DmitryChitalov commented on Урок 1. Практическое задание/task_5.py:
```diff
@@ -17,3 +17,65 @@
+
+    print(SPC.stack_size())
+
+    print(SPC.elems)
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_6.py:
```diff
@@ -17,3 +17,70 @@
+
+    print(QTBC.basetsk, ' Задач в беклоге: ', QTBC.size('base'))
+    print(QTBC.reworktsk, ' На доработку: ', QTBC.size('rework'))
+    print(QTBC.done, ' Реализовано: ', QTBC.size('done'))
```
выполнено

DmitryChitalov commented on Урок 1. Практическое задание/task_7.py:
```diff
@@ -13,3 +13,47 @@
+
+    
+if __name__ == '__main__':
+        print(pal_checker('молоко делили ледоколом'))    
```
выполнено

Sanuamba3 committed:
Достача ДЗ 1 урок Лаптев А

Лаптев Александр Сергеевич, Алгоритмы и структуры данных на Python, 9.11.22, урок 1, задание 4, ошибка в оценке сложности
